cpp/src/round/round.cu

```diff
@@ -259,8 +261,22 @@ std::unique_ptr<column> round_with(column_view const& input,
   // overflow. Under this circumstance, we can simply output a zero column because no digits can
   // survive such a large scale movement.
   if (scale_movement > cuda::std::numeric_limits<Type>::digits10) {
-    auto zero_scalar = make_fixed_point_scalar<T>(0, scale_type{-decimal_places});
-    detail::fill_in_place(out_view, 0, out_view.size(), *zero_scalar, stream);
+    if (input.nullable()) {
+      auto device_view = column_device_view::create(out_view, stream);
+      detail::copy_range(thrust::make_constant_iterator(static_cast<Type>(0)),
+                         detail::make_validity_iterator(*device_view),
+                         out_view,
+                         out_view.offset(),
+                         out_view.offset() + out_view.size(),
+                         stream);
+    } else {
+      detail::copy_range(thrust::make_constant_iterator(static_cast<Type>(0)),
+                         thrust::make_constant_iterator(false),
```

Comment:
Wait, if the input does not have null, the output will be a column of all nulls?

Reply:
Oh, you are right. It is unnecessary to set the null mask if the input doesn't have one.

Reply:
If so, then we can just use `thrust::uninitialize_fill` to fill the zero value for the output.

Reply:
Done. While, I am quite curious why we use `unititialize_fill` instead of `thrust::flll` here. Is it because `thrust::fill` will have extra cost on initializing the data before assigning the value?

Reply:
That's very interesting. I believe that for plain types they are the same. Otherwise, `unititialize_fill` calls copy constructor while `thrust::fill` call assignment operator.

Reply:
In addition, it would be nice if you can add a unit test for this case :smiley:

Reply:
That's interesting!
 The unit test for validity was added into one of the existed test blocks.